File: scan_nb.py
import json
import re

with open('CleanRAG2.ipynb', 'r', encoding='utf-8') as f:
    nb = json.load(f)

found = 0
for i, cell in enumerate(nb['cells']):
    if cell['cell_type'] != 'code':
        continue
    src = ''.join(cell['source'])
    for linenum, line in enumerate(src.split('\n')):
        if re.search(r'gsk_[A-Za-z0-9]{15,}', line):
            found += 1
            red = re.sub(r'gsk_[A-Za-z0-9]+', 'gsk_[REDACTED]', line)
            print(f'[GROQ KEY] Cell {i}, Line {linenum+1}: {red.strip()[:120]}')
        if re.search(r'qdrant', line, re.IGNORECASE) and re.search(r'["\'][A-Za-z0-9\-_]{20,}["\']', line):
            found += 1
            red = re.sub(r'(["\'])[A-Za-z0-9\-_]{20,}(["\'])', r'\1[REDACTED]\2', line)
            print(f'[QDRANT KEY] Cell {i}, Line {linenum+1}: {red.strip()[:120]}')

print(f'\nTotal exposures: {found}')
print('\n=== CELL STRUCTURE ===')
for i, cell in enumerate(nb['cells']):
    if cell['cell_type'] != 'code':
        continue
    src = ''.join(cell['source']).strip()
    if len(src) < 20:
        continue
    first = src.split('\n')[0][:130]
    print(f'Cell[{i:02d}]: {first}')

File: scan_nb.ts
import { readFileSync } from 'fs';

const nb = JSON.parse(readFileSync('CleanRAG2.ipynb', 'utf-8'));

const groqPattern = /gsk_[A-Za-z0-9]{20,}/g;
const qdrantKeyPattern = /['"][A-Za-z0-9\-_]{30,}['"]/g; // catches qdrant tokens

let foundCount = 0;

for (let i = 0; i < nb.cells.length; i++) {
  const cell = nb.cells[i];
  if (cell.cell_type !== 'code') continue;
  const src: string = Array.isArray(cell.source) ? cell.source.join('') : cell.source;
  
  const lines = src.split('\n');
  lines.forEach((line: string, lineNum: number) => {
    const hasGroq = groqPattern.test(line);
    groqPattern.lastIndex = 0;
    const hasQdrant = line.toLowerCase().includes('qdrant') && qdrantKeyPattern.test(line);
    qdrantKeyPattern.lastIndex = 0;
    if (hasGroq) {
      foundCount++;
      const redacted = line.replace(/gsk_[A-Za-z0-9]+/g, 'gsk_[REDACTED]');
      console.log(`[CREDENTIAL] Groq API key — Cell ${i}, Line ${lineNum + 1}: ${redacted.trim()}`);
    }
    if (hasQdrant) {
      foundCount++;
      const redacted = line.replace(/(['"])[A-Za-z0-9\-_]{30,}(['"])/g, '$1[REDACTED]$2');
      console.log(`[CREDENTIAL] Qdrant token   — Cell ${i}, Line ${lineNum + 1}: ${redacted.trim()}`);
    }
  });
}

console.log(`\nTotal credential exposures found: ${foundCount}`);
console.log('\n=== CELL STRUCTURE SUMMARY ===');
nb.cells.forEach((cell: any, i: number) => {
  if (cell.cell_type !== 'code') return;
  const src: string = Array.isArray(cell.source) ? cell.source.join('') : cell.source;
  if (src.trim().length < 20) return;
  const first = src.trim().split('\n')[0].substring(0, 130);
  console.log(`Cell[${i.toString().padStart(2,'0')}]: ${first}`);
});
